src/GeoPing.Api/Controllers/AccountController.cs

```diff
@@ -215,30 +222,55 @@ public IActionResult Register(string returnUrl = null)
         [HttpPost]
         [AllowAnonymous]
         [ValidateAntiForgeryToken]
-        public async Task<IActionResult> Register(RegisterViewModel model, string returnUrl = null)
+        public async Task<IActionResult> Register(RegisterUserDTO registerUser, string returnUrl = null)
         {
+            /*
+             * Checks if user with submitted email is exists
+             */
+            var context = _serviceProvider.GetRequiredService<ApplicationDbContext>();
+            if(context.Users.Any(u => u.Email == registerUser.Email))
+            {
+                return BadRequest(new OperationResult
+                {
+                    Success = false,
+                    Messages = { "Invalid username or email" }
+                });
+            }
+
             ViewData["ReturnUrl"] = returnUrl;
+            
             if (ModelState.IsValid)
             {
-                var user = new ApplicationUser { UserName = model.Email, Email = model.Email };
-                var result = await _userManager.CreateAsync(user, model.Password);
+                var user = new ApplicationUser { UserName = registerUser.UserName, Email = registerUser.Email };
+                
+                var result = await _userManager.CreateAsync(user, registerUser.Password);
                 if (result.Succeeded)
                 {
                     _logger.LogInformation("User created a new account with password.");
 
-                    var code = await _userManager.GenerateEmailConfirmationTokenAsync(user);
-                    var callbackUrl = Url.EmailConfirmationLink(user.Id, code, Request.Scheme);
-                    await _emailSender.SendEmailConfirmationAsync(model.Email, callbackUrl);
+                    //var code = await _userManager.GenerateEmailConfirmationTokenAsync(user);
+                    //var callbackUrl = Url.EmailConfirmationLink(user.Id, code, Request.Scheme);
+                    //await _emailSender.SendEmailConfirmationAsync(model.Email, callbackUrl);
 
                     await _signInManager.SignInAsync(user, isPersistent: false);
```

Comment:
This is for cookie auth, delete this string.

Reply:
done